src/circleci/rollcage/throwables.clj

```diff
@@ -0,0 +1,14 @@
+(ns circleci.rollcage.throwables)
+
+(defn cause-seq
+  [^Throwable ex]
+  (loop [^Throwable ex ex
+         result (list)]
+    (if-not ex
+      result
+      (recur (.getCause ex)
+             (conj result ex)))))
```

Comment:
This whole function could be:

```clojure
(defn cause-seq
  [^Throwable t]
  (take-while some? (iterate (memfn getCause) t)))
```

Which might be small enough to inline below, depending on your personal taste?